src/vs/workbench/contrib/scm/common/quickDiff.ts

```diff
@@ -0,0 +1,28 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { URI } from 'vs/base/common/uri';
+import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+import { IDisposable } from 'vs/base/common/lifecycle';
+
+export const IQuickDiffService = createDecorator<IQuickDiffService>('quickDiff');
+
+export interface QuickDiffProvider {
+	label: string;
+	rootUri: URI | undefined;
+	getOriginalResource(uri: URI): Promise<URI | null>;
+}
+
+export interface QuickDiff {
+	label: string;
```

Comment:
What is this label for? How would it be any different from `QuickDiffProvider.label`?

Reply:
I expect to use it when showing UX to differentiate quick diffs from different providers. It's copied here from `QuickDiffProvider` so that callers of `getQuickDiffs` have access to the label.